clients/python/test/e2e/ci/_data_classes.py

```diff
@@ -0,0 +1,172 @@
+import configparser
+from pathlib import Path
+from typing import Dict, Optional
+from urllib.parse import ParseResult, urlparse
+
+from _utils import _PYTEST_INI
+from packaging.version import Version
+from pydantic import BaseModel, field_validator, model_validator
+
+# Holds classes for passing data around between scripts.
+
+
+class ServerConfig(BaseModel):
+    """Holds data about server configuration"""
+
+    osparc_api_host: str
+    osparc_api_key: str
+    osparc_api_secret: str
+
+    @property
+    def url(self) -> ParseResult:
+        return urlparse(f"{self.osparc_api_host}")
+
+    @property
+    def key(self) -> str:
```

Comment:
minor: unless you really need key, secret, the BaseModel already provides you with properties to access directly the variables.

Reply:
see comment below why I do it like this.